src/librustc_typeck/check/generator_interior.rs

```diff
@@ -139,18 +132,29 @@ pub fn resolve_interior<'a, 'tcx>(
     // if a Sync generator contains an &'α T, we need to check whether &'α T: Sync),
     // so knowledge of the exact relationships between them isn't particularly important.
 
-    debug!("types in generator {:?}, span = {:?}", type_list, body.value.span);
+    debug!(
+        "types in generator {:?}, span = {:?}",
+        types.iter().map(|t| (t.0).ty).collect::<Vec<_>>(),
```

Comment:
We could maybe just print `types` in this debug log rather than getting only the types, it'll be a bit more output but avoids needing to collect into a `Vec` just for this log.